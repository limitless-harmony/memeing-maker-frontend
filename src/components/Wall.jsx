import React from 'react';
import styled from 'styled-components';

import { calculateRem } from 'styles';
import { dark } from 'styles/colors';
import MemeContainer from 'components/MemeContainer';
import Tagline from 'components/Tagline';

const MemeWall = ({ wall }) => {
  return (
    <Container>
      <Tagline />
      {wall && (
        <>
          <WallTitle>{wall.name}</WallTitle>
          {wall.memes.length ? (
            <MemeContainer memes={wall.memes} />
          ) : (
            <SectionHeading>No meme available in this wall yet!</SectionHeading>
          )}
        </>
      )}
    </Container>
  );
};

const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: ${calculateRem(18)} auto ${calculateRem(40)};
  width: 100%;
`;

const WallTitle = styled.div`
  margin: ${calculateRem(5)} auto ${calculateRem(50)};
  padding: ${calculateRem(14)} ${calculateRem(25)};
  border: ${calculateRem(3)} solid ${dark};
  border-radius: ${calculateRem(7)};
  box-sizing: border-box;
  box-shadow: ${calculateRem(2)} ${calculateRem(4)} ${calculateRem(4)} 0 ${dark};
  text-align: center;
  font-size: ${calculateRem(23)};
  line-height: ${calculateRem(27)};
  width: 80%;
`;

const SectionHeading = styled.div`
  font-size: ${calculateRem(18)};
  margin: ${calculateRem(16)} auto;
`;
export default MemeWall;
