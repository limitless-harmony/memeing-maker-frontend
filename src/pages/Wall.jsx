import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import styled from 'styled-components';

import { getAWall, getWalls } from 'actions/wall';
import WallContainer from 'components/WallContainer';
import Wall from 'components/Wall';
import { calculateRem } from 'styles';

export class MemeWall extends Component {
  async componentDidMount() {
    const { match } = this.props;
    const {
      params: { id },
    } = match;
    this.fetchData(id);
  }

  componentDidUpdate(prevProps) {
    const {
      match: { params },
    } = prevProps;
    const {
      match: {
        params: { id },
      },
    } = this.props;
    if (id !== params.id) {
      this.fetchAWall(id);
    }
  }

  fetchAWall = async wallId => {
    const { actions } = this.props;
    await actions.getAWall(wallId);
  };

  fetchAllWalls = async () => {
    const { actions } = this.props;
    await actions.getWalls();
  };

  fetchData = async id => {
    await this.fetchAWall(id);
    await this.fetchAllWalls();
  };

  render() {
    const { wall, walls, history } = this.props;
    return (
      <>
        <Wall wall={wall} />
        <MoreWalls>More Meme Walls</MoreWalls>
        <WallContainer walls={walls} history={history} />
      </>
    );
  }
}

const MoreWalls = styled.div`
  margin: 0 0 ${calculateRem(10)};
  font-size: ${calculateRem(18)};
  text-align: center;
`;

const mapStateToProps = state => ({
  wall: state.wall.current,
  walls: state.wall.walls,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({ getAWall, getWalls }, dispatch),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MemeWall);
