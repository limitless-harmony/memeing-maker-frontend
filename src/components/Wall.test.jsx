import React from 'react';
import { shallow } from 'enzyme';
import Wall from 'components/Wall';

describe('Testing the Featured page component', () => {
  const wall = {
    name: 'Example wall',
    memes: [],
  };
  it('renders without crashing', () => {
    const wrapper = shallow(<Wall wall={wall} />);
    expect(wrapper.find('div'));
    expect(wrapper).toMatchSnapshot();
  });
});
