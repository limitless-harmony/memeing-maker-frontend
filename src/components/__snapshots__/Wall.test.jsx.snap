// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Featured page component renders without crashing 1`] = `
<styled.div>
  <Tagline />
  <styled.div>
    Example wall
  </styled.div>
  <styled.div>
    No meme available in this wall yet!
  </styled.div>
</styled.div>
`;
